crates/goose-cli/src/main.rs

```diff
@@ -145,15 +147,24 @@ async fn main() -> Result<()> {
         }
         Some(Command::Run {
             instructions,
+            input_text,
             profile,
             session,
             resume,
         }) => {
-            let file_name =
-                instructions.expect("Instruction file is required (--instructions <file_path>)");
-            let file_path = std::path::Path::new(&file_name);
-            let contents = std::fs::read_to_string(file_path).expect("Failed to read the file");
-
+            let contents = if let Some(file_name) = instructions {
+                let file_path = std::path::Path::new(&file_name);
+                std::fs::read_to_string(file_path).expect("Failed to read the instruction file")
+            } else if let Some(input_text) = input_text {
+                input_text
+            } else {
+                use std::io::{self, Read};
+                let mut stdin = String::new();
+                io::stdin()
+                    .read_to_string(&mut stdin)
+                    .expect("Failed to read from stdin");
```

Comment:
it sounds easy, but i think the way to do it would be to spawn a thread and then poll the channel. I'll skip this for now.

 https://stackoverflow.com/questions/30012995/how-can-i-read-non-blocking-from-stdin